### DISCUSSION TITLE
Underline plugin

### DISCUSSION THREAD

**Discussion by VityaSchel** [remarkjs/remark#652] [Ideas] [answered]
I really need underline for markdown

**Comment by wooorm** [comment, marked as answer]
You can find existing plugins here: https://github.com/remarkjs/remark/blob/main/doc/plugins.md#list-of-plugins.
Here’s a guide on how to create your own plugins: https://unifiedjs.com/learn/guide/create-a-plugin/.
You could perhaps use remark-directive for this: https://github.com/remarkjs/remark-directive.

**Reply by VityaSchel** [reply]
There is no underline plugin and remark-directive isn't for it. Please don't mark as solved

**Reply by ChristianMurphy** [reply]
it is solved by https://github.com/Darkhax/remark-underline#readme, if it doesn't meet your needs the instructions provided show how to make a new plugin

**Comment by ChristianMurphy** [comment]
Doing a quick web search for "remark underline" turns up https://github.com/Darkhax/remark-underline#readme

**Reply by VityaSchel** [reply]
It does not work with new version of parser

**Reply by VityaSchel** [reply]
I wouldn't post an issue if there was this easy solution, would I?
[https://github.com/Darkhax/remark-underline/issues/2](https://github.com/Darkhax/remark-underline/issues/2)

**Reply by ChristianMurphy** [reply]
Easy is a matter of perspective. :man_shrugging: 

The answer is straight forward.
https://github.com/Darkhax/remark-underline#readme can be used with remark 12.
or you can resolve Darkhax/remark-underline#2 so it can work with remark 13+.

If those don't suit your needs, @wooorm provided several alternatives https://github.com/remarkjs/remark/discussions/652#discussioncomment-437332
Including creating a new plugin or using directives.

**Comment by puopg** [comment]
Possible example for anyone interested.

https://codesandbox.io/s/bold-pond-4ypph4?file=/src/App.js
